org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandler.java

```diff
@@ -198,6 +199,9 @@ private void publishMarkers(IProject project, IMarker[] markers) throws CoreExce
 			if (monitor != null && monitor.isCanceled()) {
 				throw new OperationCanceledException();
 			}
+			if (!ProjectUtils.isJavaProject(project)) {
```

Comment:
pom projects (so not java projects) might have some maven related errors needing to be displayed